src/gfx.cpp

```diff
@@ -1445,20 +1438,75 @@ void RedrawScreenRect(int left, int top, int right, int bottom)
 	VideoDriver::GetInstance()->MakeDirty(left, top, right - left, bottom - top);
 }
 
+static std::vector<Rect> _dirty_viewport_occlusions;
+static const Viewport *_dirty_viewport;
+static NWidgetDisplay _dirty_viewport_disp_flags;
+
+static void DrawDirtyViewport(uint occlusion, int left, int top, int right, int bottom)
+{
+	for(; occlusion < _dirty_viewport_occlusions.size(); occlusion++) {
+		const Rect &occ = _dirty_viewport_occlusions[occlusion];
+		if (right <= occ.left || bottom <= occ.top || left >= occ.right || top >= occ.bottom) {
+			/* No intersection. */
+			continue;
+		}
+
+		int middle;
+		if (left < (middle = occ.left)) {
```

Comment:
dirty. `int middle = foo;` should go inside the if block.

Reply:
I forgot you can do that in C :+1:

Reply:
Changing the line to `if (left < (int middle = occ.left))) {` results in
```
error: expected primary-expression before 'int'
error: expected ')' before 'int'
```
:disappointed: